pkg/git/repository.go

```diff
@@ -229,6 +247,41 @@ var clone = func(ctx context.Context, opts *CloneOptions) (*repo, error) {
 	return repo, nil
 }
 
+var createRepo = func(ctx context.Context, opts *CloneOptions) (string, error) {
+	host, orgRepo, _, _, _, _, _ := util.ParseGitUrl(opts.Repo)
+	providerType := opts.Provider
+	if providerType == "" {
+		u, err := url.Parse(host)
+		if err != nil {
+			return "", err
+		}
+
+		providerType = strings.TrimSuffix(u.Hostname(), ".com")
```

Comment:
I wonder how good is this heuristic..
I would print a warning message here: `--provider not specified, assuming provider from url: %s` where `%s` is the guessed provider name.